msg_queue.go

```diff
@@ -34,6 +95,36 @@ func (m *msgQueue) readMessage(st State, current *View) *MessageReq {
 	return msg
 }
 
+func readValues(queue *msgQueueImpl, current *View) (*MessageReq, []*MessageReq) {
```

Comment:
Just because of that single if/else branch which checks `PbftState` we are duplicating rest of the code. 
Not a fan of this approach, to be honest and to me it is not in line with good coding practice.

Instead I think we should extract it into the function and have different handler defined for `RoundChangeState` and other `PbftState` values.